packages/react/src/hooks/useVoiceAssistant.ts

```diff
@@ -42,8 +43,19 @@ const state_attribute = 'lk.agent.state';
  * @beta
  */
 export function useVoiceAssistant(): VoiceAssistant {
-  const agent = useRemoteParticipants().find((p) => p.kind === ParticipantKind.AGENT);
-  const audioTrack = useParticipantTracks([Track.Source.Microphone], agent?.identity)[0];
+  const agent = useRemoteParticipants().find(
+    (p) => p.kind === ParticipantKind.AGENT && !('lk.publish_on_behalf' in p.attributes),
+  );
+  const worker = useRemoteParticipants().find(
+    (p) =>
+      p.kind === ParticipantKind.AGENT && p.attributes['lk.publish_on_behalf'] === agent?.identity,
+  );
+  const agentAudioTrack = useParticipantTracks([Track.Source.Microphone], agent?.identity)[0];
```

Comment:
similar notion as above, it's generally more efficient to filter on `useParticipantTracks(['microphone', 'camera'])` than calling it multiple times.